packages/iov-stream/src/concat.spec.ts

```diff
@@ -0,0 +1,217 @@
+// tslint:disable:readonly-array no-submodule-imports
+import { Producer, Stream } from "xstream";
+
+import { concat } from "./concat";
+
+function producerIsStopped(): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, 10));
+}
+
+describe("concat", () => {
+  it("can concat 0 streams", done => {
+    const concatenatedStream = concat();
+    const expected: string[] = [];
+
+    concatenatedStream.addListener({
+      next: value => expect(value).toEqual(expected.shift()!),
+      complete: () => {
+        expect(expected.length).toEqual(0);
+        done();
+      },
+      error: done.fail,
+    });
+  });
+
+  it("can concat 1 streams", done => {
+    const stream1 = Stream.of("1", "2", "3");
+    const concatenatedStream = concat(stream1);
+    const expected = ["1", "2", "3"];
+
+    concatenatedStream.addListener({
+      next: value => expect(value).toEqual(expected.shift()!),
+      complete: () => {
+        expect(expected.length).toEqual(0);
+        done();
+      },
+      error: done.fail,
+    });
+  });
+
+  it("can concat 2 streams", done => {
+    const stream1 = Stream.of("1", "2", "3");
+    const stream2 = Stream.of("a", "b", "c");
+    const concatenatedStream = concat(stream1, stream2);
+    const expected = ["1", "2", "3", "a", "b", "c"];
+
+    concatenatedStream.addListener({
+      next: value => expect(value).toEqual(expected.shift()!),
+      complete: () => {
+        expect(expected.length).toEqual(0);
+        done();
+      },
+      error: done.fail,
+    });
+  });
+
+  it("can concat 3 streams", done => {
+    const stream1 = Stream.of("1", "2", "3");
+    const stream2 = Stream.of("a", "b", "c");
+    const stream3 = Stream.of("X", "Y", "Z");
+    const concatenatedStream = concat(stream1, stream2, stream3);
+    const expected = ["1", "2", "3", "a", "b", "c", "X", "Y", "Z"];
+
+    concatenatedStream.addListener({
+      next: value => expect(value).toEqual(expected.shift()!),
+      complete: () => {
+        expect(expected.length).toEqual(0);
+        done();
+      },
+      error: done.fail,
+    });
+  });
+
+  it("changes output order when order of streams switch", done => {
+    const stream1 = Stream.of("1", "2", "3");
+    const stream2 = Stream.of("a", "b", "c");
+    const concatenatedStream = concat(stream2, stream1);
+    const expected = ["a", "b", "c", "1", "2", "3"];
+
+    concatenatedStream.addListener({
+      next: value => expect(value).toEqual(expected.shift()!),
+      complete: () => {
+        expect(expected.length).toEqual(0);
+        done();
+      },
+      error: done.fail,
+    });
+  });
+
+  it("changes output order when order of streams switch", done => {
```

Comment:
this test case is duplicated above.
either remove it or make it unique

Reply:
Oh, indeed a copy/paste error. Thanks!